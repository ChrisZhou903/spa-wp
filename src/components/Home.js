import React, { PureComponent } from 'react'
import { object, func } from 'prop-types'

import './Home.css'
import logo from '../assets/images/logo.svg'

class Home extends PureComponent {
  static propTypes = {
    posts: object.isRequired,
    fetchPosts: func.isRequired,
  }

  componentDidMount() {
    const { fetchPosts } = this.props

    fetchPosts()
  }

  render() {
    const { posts } = this.props
    const isEmpty = Object.keys(posts).length === 0

    return (
      <div className="App container">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>
            Welcome to WP SPA
            {isEmpty && 'There is no posts yet.'}
          </h2>
        </div>
      </div>
    )
  }
}

export default Home
