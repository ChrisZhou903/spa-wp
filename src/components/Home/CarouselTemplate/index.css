.home-carousel .ant-carousel {
  margin-top: 64px;
}

.home-carousel .slick-track {
  height: 100vh;
}

.home-carousel .slick-slide {
  position: relative;
  background-size: cover;
  background-position: 50% 50%;
  overflow: hidden;
}

.home-carousel .slick-slide:before {
  display: block;
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, .3);
}

.home-carousel__slogan {
  width: 100%;
  position: absolute;
  left: 0;
  top: 50%;
  transform: translate(0, -50%);
  text-align: center;
}

.home-carousel__slogan h1,
.home-carousel__slogan h3 {
  color: #fff;
}
