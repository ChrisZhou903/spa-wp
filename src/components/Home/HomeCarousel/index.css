.ant-carousel {
  margin-top: 64px;
}

.slick-track {
  height: 80vh;
}

.slick-slide {
  position: relative;
  background-size: cover;
  background-position: center;
  overflow: hidden;
}

.slick-slide:before {
  display: block;
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, .3);
}

.slogan {
  width: 100%;
  position: absolute;
  left: 0;
  top: 50%;
  transform: translate(0, -50%);
  text-align: center;
}

.slogan h1,
.slogan h3 {
  color: #fff;
}
