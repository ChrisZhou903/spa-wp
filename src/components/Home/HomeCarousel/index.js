import React from 'react'
import { array, object } from 'prop-types'
import { Carousel } from 'antd'

import './index.css'

const HomeCarousel = ({
  site,
  carousel,
}) => (
  carousel.length > 0 &&
  <Carousel
    draggable
    autoplay
    autoplaySpeed="5000"
  >
    {carousel.map(media => (
      <div key={media.id} style={{ backgroundImage: `url(${media.sourceUrl})` }}>
        <div className="slogan">
          <h1>{site.name}</h1>
          <h3>{site.description}</h3>
        </div>
      </div>
    ))}
  </Carousel>
)

HomeCarousel.propTypes = {
  site: object.isRequired,
  carousel: array.isRequired,
}

export default HomeCarousel
