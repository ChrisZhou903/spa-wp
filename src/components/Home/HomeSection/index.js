import React from 'react'
import { object } from 'prop-types'

const HomeSection = ({
  content,
}) => (
  <section
    className="section"
    dangerouslySetInnerHTML={
      { __html: content }
    }
  />
)

HomeSection.propTypes = {
  content: object.isRequired,
}

export default HomeSection
