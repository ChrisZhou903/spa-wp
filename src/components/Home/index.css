.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  height: 150px;
  padding: 20px;
}

.App-intro {
  font-size: large;
}

.add-vote {
  margin-top: 1rem;
}

.App-votes {
  margin-top: 2rem;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
