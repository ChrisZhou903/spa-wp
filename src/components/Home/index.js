import React, { PureComponent } from 'react'
import { number, object, array, func } from 'prop-types'
import Loadable from 'react-loadable'

import Loading from '../Loading'
import './index.css'

const CarouselTemplate = Loadable({
  loader: () => import('../CarouselTemplate'),
  loading: Loading,
})

const SectionTemplate = Loadable({
  loader: () => import('../SectionTemplate'),
  loading: Loading,
})

class Home extends PureComponent {
  static propTypes = {
    pageId: number,
    page: object.isRequired,
    homeCarousel: array.isRequired,
    pageSection: array.isRequired,
    fetchSinglePage: func.isRequired,
  }

  static defaultProps = {
    pageId: 0,
  }

  componentDidMount() {
    const { pageId, page, fetchSinglePage } = this.props

    if (pageId && Object.keys(page).length === 0) {
      fetchSinglePage(pageId)
    }
  }

  render() {
    const { pageSection, homeCarousel } = this.props

    return (
      <div className="app-home">
        <CarouselTemplate carousel={homeCarousel} />
        <div className="app-home__content">
          {
            pageSection.map((section, index) =>
              (<SectionTemplate
                key={index} // eslint-disable-line
                className={`app-section_${index + 1}`}
                section={section}
              />),
            )
          }
        </div>
      </div>
    )
  }
}

export default Home
