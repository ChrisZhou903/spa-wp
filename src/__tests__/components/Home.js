import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import Home from '../../components/Home'

const initProps = {
  pageId: 0,
  page: {},
  posts: {},
  homeCarousel: [],
  pageSection: [],
  fetchPosts: jest.fn(),
  fetchSinglePage: jest.fn(),
}
const setup = (props = initProps) => {
  const component = <Home {...props} />
  const componentWithRouter = (
    <MemoryRouter>
      <Home {...props} />
    </MemoryRouter>
  )
  const wrapper = shallow(component)

  return {
    component,
    componentWithRouter,
    wrapper,
  }
}

describe('Home component', () => {
  it('should render without errors', () => {
    const { wrapper } = setup()

    expect(wrapper).toBeTruthy()
  })

  it('should render correct things', () => {
    const { wrapper } = setup()

    expect(wrapper.find('.app-home').length === 1).toBe(true)
  })

  it("should render right thing and don't change unexpected", () => {
    const { componentWithRouter } = setup()
    const tree = renderer.create(componentWithRouter).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
