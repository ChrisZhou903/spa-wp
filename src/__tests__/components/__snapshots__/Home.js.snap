// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render right thing and don't change unexpected 1`] = `
<div
  className="app-home"
>
  <div
    className="loading"
  >
    <div
      className="ant-spin ant-spin-spinning ant-spin-show-text"
    >
      <span
        className="ant-spin-dot"
      >
        <i />
        <i />
        <i />
        <i />
      </span>
      <div
        className="ant-spin-text"
      >
        Loading...
      </div>
    </div>
  </div>
  <div
    className="app-home__content"
  />
</div>
`;
